.github/workflows/build_wheels.yml

```diff
@@ -193,29 +196,232 @@ jobs:
         GITHUB_BASE_REF=$GITHUB_BASE_REF
         EOF
         echo $(cat github_action_info)
-        gsutil cp -a public-read github_action_info ${{ env.GCP_PATH }}
+        gsutil cp -a public-read github_action_info ${{ env.GCP_WHEELS_STAGING_PATH }}
     - name: Upload GitHub event file to GCS bucket
-      run: gsutil cp -a public-read ${GITHUB_EVENT_PATH} ${{ env.GCP_PATH }}
+      run: gsutil cp -a public-read ${GITHUB_EVENT_PATH} ${{ env.GCP_WHEELS_STAGING_PATH }}
 
   list_files_on_gcs:
     name: List files on Google Cloud Storage Bucket
-    needs: upload_wheels_to_gcs
+    needs: upload_python_sdk_wheels_to_gcs
     runs-on: ubuntu-latest
-    if: github.repository_owner == 'apache' && github.event_name != 'pull_request'
+    if: github.repository_owner == 'apache' && (github.event_name == 'push' || github.event_name == 'schedule')
     steps:
     - name: Authenticate on GCP
       uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
       with:
         service_account_email: ${{ secrets.GCP_SA_EMAIL }}
         service_account_key: ${{ secrets.GCP_SA_KEY }}
     - name: List file on Google Cloud Storage Bucket
-      run: gsutil ls "${{ env.GCP_PATH }}*"
+      run: gsutil ls "${{ env.GCP_WHEELS_STAGING_PATH }}*"
+
+  python_unit_tests:
+    name: 'Python Unit Tests'
+    runs-on: ${{ matrix.os }}
+    if: github.event_name == 'push' || github.event_name == 'schedule'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        params: [
+          {"py_ver": "3.5", "tox_env": "py35"},
+          {"py_ver": "3.6", "tox_env": "py36"},
+          {"py_ver": "3.7", "tox_env": "py37"},
```

Comment:
@tvalentyn - Should we add 3.8?

Reply:
Yes

Reply:
Do all tests against all versions run in parallel?

Reply:
I checked running different version of python in one job but time saved by that is very small (~30min vs ~26min) but is much more harder to figure out what went wrong in long logs:
https://github.com/TobKed/beam/runs/901872919?check_suite_focus=true
Jobs will run in parallel manner (if enough runners available at the moment)